lib/HelloSignResource.js

```diff
@@ -334,7 +334,11 @@ HelloSignResource.prototype = {
             // submit as form
             var form = new FormData();
             for(var i=0; i<data.files.length;i++) {
-                form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
+                if (typeof data.files[i] === 'string') {
+                    form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
+                } else if (Buffer.isBuffer(data.files[i])) {
+                    form.append("file[" + i + "]", data.files[i]);
+                }
```

Comment:
Hey @drbobbeaty! Thanks for the PR :smiley:

Glancing through this, I actually think we may want to swap these two conditions:

```suggestion
                if (Buffer.isBuffer(data.files[i])) {
                    form.append("file[" + i + "]", data.files[i]);
                } else {
                    form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
                }
```

In the past we never checked the type of `data.files[i]`, but this new code is specifically checking if it's a string. I think to be safe, we should instead check if it's a buffer first, and then fall back to the old code as a catch-all.

Reply:
@nathanbuchar  I made the requested change, are there any _other_ changes that are needed to unblock this?